packages/editor/src/components/Editor.js

```diff
@@ -135,27 +102,50 @@ class BlockEditor extends Component {
   }
 
   render() {
+    const { plugins, theme, components } = this.props
+    const allComponents = {
+      ...defaultBlocks,
+      ...components
+    }
+    const context = {
+      components: allComponents
+    }
+
     return (
       <div style={{ minHeight: '100vh' }}>
-        <Editor
-          {...this.props}
-          ref={editor => (this.editor = editor)}
-          schema={schema}
-          placeholder="Write some MDX..."
-          plugins={plugins}
-          value={this.state.value}
-          onChange={this.handleChange}
-          onKeyDown={this.handleKeyDown}
-          onPaste={this.handlePaste}
-          renderEditor={(_props, _editor, next) => {
-            const children = next()
-
-            return <>{children}</>
-          }}
-        />
+        <Context.Provider value={context}>
+          <Editor
+            {...this.props}
+            ref={editor => (this.editor = editor)}
+            components={allComponents}
+            theme={theme}
+            schema={schema}
+            placeholder="Write some MDX..."
+            plugins={plugins}
+            value={this.state.value}
+            onChange={this.handleChange}
+            onKeyDown={this.handleKeyDown}
+            onPaste={this.handlePaste}
+          />
+        </Context.Provider>
       </div>
     )
   }
 }
 
+BlockEditor.defaultProps = {
+  components: {},
+  theme: defaultTheme,
+  plugins: defaultPlugins,
+  renderEditor: (props, editor, next) => {
```

Comment:
If you wanted to customize the toolbar from the outside, you could pass a custom `renderEditor` prop – eventually we'd expose the root `Toolbar` component and buttons as an exports